ipn/local.go

```diff
@@ -454,7 +460,7 @@ func (b *LocalBackend) updateFilter(netMap *controlclient.NetworkMap, prefs *Pre
 	// Be conservative while not ready.
 	shieldsUp := prefs == nil || prefs.ShieldsUp
 
-	changed := deepprint.UpdateHash(&b.filterHash, packetFilter, advRoutes, shieldsUp)
+	changed := deepprint.UpdateHash(&b.filterHash, netMap.Addresses, packetFilter, advRoutes, shieldsUp)
```

Comment:
Or move this after the UpdateHash so there's no early return. You'll need to do something like:

```go
haveNetmap := netMap != nil
var addesses ...
var packetFilter ..
if haveNetmap {
   addresses = netMap.Addresses
   packetFilter = ...
}
changed := UpdateHash(&b.filterHash, haveNetMap, addresses, packetFilter, advRoutes, shieldsUp)
```

Reply:
(idea being then you don't need a big scary comment at top and there's one fewer way out of the func)

Reply:
Like this?